src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newArray = [];
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        newArray[i] = '';
+      } else {
+        newArray[i] = this[i];
```

Comment:
[CODE STYLE] - avoid using nested if statements, we are sure that you can do it without them 
Use for example ternary operator like 2 strings lower or try to find another condition for `If`